sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -870,6 +870,7 @@ case class ArrayMin(child: Expression) extends UnaryExpression with ImplicitCast
 
   override protected def nullSafeEval(input: Any): Any = {
     var min: Any = null
+    val ordering = TypeUtils.getInterpretedOrdering(dataType)
```

Comment:
We can't do this, `nullSafeEval` is called for every input record

Reply:
Ok, I will fix it.